src/main/scala/io/iohk/ethereum/network/discovery/PeerDiscoveryManager.scala

```diff
@@ -95,10 +94,11 @@ class PeerDiscoveryManager(
       case ServerStatus.Listening(address) =>
         val from = Endpoint.makeEndpoint(address, getTcpPort)
         val ping = Ping(ProtocolVersion, from, toEndpoint, expirationTimestamp)
-        getPacketData(ping).foreach{ key =>
+        val packet = encodePacket(ping, nodeStatusHolder.get().key)
+        getPacketData(packet).foreach{ key =>
```

Comment:
Maybe it would be better to not create `Packet` object ? I mean now it is created in :
https://github.com/input-output-hk/mantis/pull/608/files#diff-9fc6e18903892ffc78f87ca4811bb1c5R101 and
https://github.com/input-output-hk/mantis/pull/608/files#diff-9fc6e18903892ffc78f87ca4811bb1c5R117
and it is possibly high cost operation ( each packet has `val nodeId` which is calculated from public key).
And all we need is two `ByteStrings` which we can recover from encoded packet in line `97` like:
`val packetMdc          = packet.take(Packet.MdcLength)` and
`val packetDataHash = crypto.kec256(packet.drop(Packet.DataOffset))`.

Second option is to make `nodeId` a `lazy val`, but then we would need to change `decodePacket` as now it is using `.get` on `nodeId` to signalize error.